File: Python/problem009.py
#! /usr/bin/env python3
"""
Problem 9 - Project Euler
http://projecteuler.net/index.php?section=problems&id=9
"""
import math


def getPythagoreanTriplet(a):
    for b in reversed(range(1, a)):
        c = math.sqrt(a**2 - b**2)
        if c.is_integer():
            return (a, b, int(c))
    else:
        return (0, 0, 0)


if __name__ == "__main__":
    x = 5
    while True:
        a, b, c = getPythagoreanTriplet(x)
        if a > 0 and 1000 % (a + b + int(c)) == 0:
            q = 1000 // (a + b + int(c))
            a, b, c = map(lambda x: x * q, [a, b, c])
            print("abc=" + str(a * b * c) + " (a,b,c)=" + str(list([a, b, c])))
            break
        elif (a + b + c) > 1000:
            print(str(list([a, b, c])))
            break
        x += 1

File: Ruby/problem009.rb
#! /usr/bin/env ruby
=begin
Problem 9 - Project Euler
http://projecteuler.net/index.php?section=problems&id=9
=end
(5..1000).each {|c|
    (4..c).each {|b|
        (3..b).each {|a|
            if a*a + b*b == c*c && 1000 % (a+b+c) == 0
                puts a*b*c*(1000.quo(a+b+c).to_i).pow(3)
                return
            end
        }
    }
}
